src/main/java/org/verdictdb/jdbc41/VerdictStreamResultSet.java

```diff
@@ -0,0 +1,1329 @@
+package org.verdictdb.jdbc41;
+
+import org.apache.commons.lang3.time.FastDateFormat;
+import org.verdictdb.VerdictSingleResult;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.*;
+import java.sql.Date;
+import java.util.*;
+import java.util.concurrent.BlockingDeque;
+import java.util.concurrent.LinkedBlockingDeque;
+
+/**
+ * Created by: Shucheng Zhong on 9/12/18
+ * ResultSet type for stream select query when VerdictStatement.sql() is called.
+ * <p>
+ * It contains a blocking queue: queryResults. When user call next(), if no queryResult available,
+ * it will try to take one from queryResults. Blocked if no results are returned from VerdictResultStream.
+ * The first column is verdictStreamSequenceColumn, which should be int class that specify the block sequence number
+ */
+public class VerdictStreamResultSet extends VerdictResultSet {
+
+  private static final String verdictStreamSequenceColumn = "seq";
+
+  boolean isCompleted = false;
+
+  private long rowIndex = 0;
+
+  private HashMap<String, Integer> colNameIdx = new HashMap<>();
+
+  BlockingDeque<VerdictSingleResult> queryResults = new LinkedBlockingDeque<>();
+
+  VerdictSingleResult queryResult;
+
+  private int lastQueryResultIndex = 0;
+
+  private VerdictStreamResultSetMetaData metadata;
+
+  public VerdictStreamResultSet() {
+    super();
+  }
+
+  public VerdictStreamResultSet(VerdictSingleResult queryResult) {
+    super(queryResult);
+    queryResults.add(queryResult);
+  }
+
+  public void appendSingleResult(VerdictSingleResult queryResult) {
+    queryResults.add(queryResult);
+  }
+
+  /**
+   * return true if getting all SingleResults
+   */
+  void setCompleted() {
+    isCompleted = true;
+  }
+
+  @Override
+  public boolean absolute(int row) throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public void afterLast() throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public void beforeFirst() throws SQLException {
+    rowIndex = 0;
+    lastQueryResultIndex = 0;
+    //    throw new SQLFeatureNotSupportedException();
+  }
+
+  @Override
+  public void cancelRowUpdates() throws SQLException {
+    throw new SQLFeatureNotSupportedException();
+  }
+
+  private void checkIndex(int index) throws SQLException {
+    if (index < 2) {
+      throw new SQLException("Column index must be a positive integer.");
```

Comment:
I understand the role, but the message itself is incorrect since the getBlob(1) will call this function, which will then throw this message.

Let's do something like:
```java
if (index == 1) {
  throw new SQLException("The first column of a streaming query result is the sequence number of each result set; thus, the values can be retrieved only in the types compatible with integer.");
} else if (index < 1) {
  throw new SQLException("A column index must be a positive integer.");
}
```